### ISSUE TITLE
[FEATURE] Add proxy settings to request config

### ISSUE THREAD

**Issue by dees91** [skrapeit/skrape.it#84]
I would like to make scrape requests via proxy gateway. 

`WebClientOptions` from htmlunit library which is used by skrape.it has `setProxyConfig` method so it is doable.

My proposition:
```
    private fun WebClient.withOptions() = apply {
        cssErrorHandler = SilentCssErrorHandler()
        ajaxController = NicelyResynchronizingAjaxController()
        createCookies()
        addRequestHeader("User-Agent", request.userAgent)
        if (request.authentication != null) {
            addRequestHeader("Authorization",request.authentication!!.toHeaderValue())
        }
        request.headers.forEach {
            addRequestHeader(it.key, it.value)
        }
        @Suppress("MagicNumber") waitForBackgroundJavaScript(10_000)
        options.apply {
            timeout = request.timeout
            isRedirectEnabled = request.followRedirects
            maxInMemory = 0
            isUseInsecureSSL = request.sslRelaxed
            isCssEnabled = false
            isPopupBlockerEnabled = true
            isDownloadImages = false
            isThrowExceptionOnScriptError = false
            isThrowExceptionOnFailingStatusCode = false
            isPrintContentOnFailingStatusCode = false
            historySizeLimit = 0
            historyPageCacheLimit = 0
            proxyConfig = request.proxyConfig //<-- here is how it may looks like
        }
    }
```

Thanks!

**Comment by christian-draeger** [comment]
we need. to find a way to test this properly. as far  as i can see wiremock supports [proxying](http://wiremock.org/docs/proxying/). i will have a look tomorrow how to implement this.

current usageexample can be found here [here](https://github.com/skrapeit/skrape.it/blob/be56c07aad5f193ee25ca7d5f66ac46e6a824e6a/src/test/kotlin/it/skrape/ExperimentalDslTest.kt#L194) and is availible in our [latest jitpack release](https://github.com/skrapeit/skrape.it#using-bleeding-edge-features-before-official-release).

will make an official release soon-ish

**Comment by cvb941** [comment]
If someone needs to use a proxy right now, I use these calls to setup the proxy on the JVM itself (applies to all connections)

```java
System.setProperty("http.proxyHost", www.host.com)
System.setProperty("http.proxyPort", 1234)
System.setProperty("http.nonProxyHosts", "localhost|127.*|[::1]|google.com")
```

**Comment by autaut03** [comment]
Hey @christian-draeger. Can we get this published? It's been 8 months since last release on Maven and jitpack doesn't work (401 unauthorized errors) :/

**Comment by christian-draeger** [comment]
Hey,
I just published version 1.0.0-alpha7 right now
Sorry for inconvenience. I will try to work on the module split soon and thereby release 1.0.0 final
